libnavigator/src/test/java/com/mapbox/navigation/navigator/internal/router/RouterInterfaceDelegateTest.kt

```diff
@@ -0,0 +1,264 @@
+package com.mapbox.navigation.navigator.internal.router
+
+import com.mapbox.api.directions.v5.DirectionsCriteria
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.bindgen.Expected
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.route.RouteRefreshCallback
+import com.mapbox.navigation.base.route.RouteRefreshError
+import com.mapbox.navigation.base.route.Router
+import com.mapbox.navigation.base.route.RouterCallback
+import com.mapbox.navigation.base.route.RouterFailure
+import com.mapbox.navigation.base.route.RouterOrigin
+import com.mapbox.navigator.RouterError
+import com.mapbox.navigator.RouterErrorType
+import com.mapbox.navigator.RouterRefreshCallback
+import io.mockk.CapturingSlot
+import io.mockk.Runs
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.mockkStatic
+import io.mockk.slot
+import io.mockk.unmockkStatic
+import junit.framework.Assert.assertEquals
+import junit.framework.Assert.assertNotNull
+import junit.framework.Assert.assertTrue
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import java.net.URL
+
+class RouterInterfaceDelegateTest {
+
+    private lateinit var mockRouter: Router
```

Comment:
Shared fields between tests makes them harder to read. The more tests you have the less % of the fields and setup logic shared between them, which makes complex to understand what a single test do. Let's move it to the individual tests

Reply:
I think there's no silver bullet, we can use shared filed for some cases like this: Router is used in every single test and it remove one line in test function